includes/validation/class-amp-invalid-url-post-type.php

```diff
@@ -583,92 +626,78 @@ public static function output_custom_column( $column_name, $post_id ) {
 						}
 					}
-					echo implode( ', ', $items ); // WPCS: XSS OK.
+				}
+				if ( ! empty( $items ) ) {
+					echo implode( ',<br/>', $items ); // WPCS: XSS OK.
 				} else {
 					esc_html_e( '--', 'amp' );
 				}
 				break;
 			case AMP_Validation_Error_Taxonomy::SOURCES_INVALID_OUTPUT:
 				if ( isset( $error_summary[ AMP_Validation_Error_Taxonomy::SOURCES_INVALID_OUTPUT ] ) ) {
-					$sources = array();
-					foreach ( $error_summary[ AMP_Validation_Error_Taxonomy::SOURCES_INVALID_OUTPUT ] as $type => $names ) {
-						foreach ( array_unique( $names ) as $name ) {
-							$sources[] = sprintf( '%s: <code>%s</code>', esc_html( $type ), esc_html( $name ) );
+					$sources = $error_summary[ AMP_Validation_Error_Taxonomy::SOURCES_INVALID_OUTPUT ];
+					$output  = array();
+
+					if ( isset( $sources['plugin'] ) ) {
+						$output[]                  = '<div class="source">';
+						$count                     = count( array_unique( $sources['plugin'] ) );
+						$sources_container_classes = 'sources-container sources-plugins';
+						if ( 1 === $count ) {
+							$output[] = sprintf( '<span class="dashicons dashicons-admin-plugins"></span></span><strong>%s</strong><br/>', esc_html__( 'Plugin', 'amp' ) );
+						} else {
+							$output[]                   = sprintf( '<span class="dashicons dashicons-admin-plugins"></span><strong>%s (%d)</strong><span class="dashicons toggle-sources dashicons-arrow-down"></span>', esc_html__( 'Plugins', 'amp' ), $count );
+							$sources_container_classes .= ' collapsed';
 						}
+						$output[] = '<div class="' . $sources_container_classes . '">';
+						$output[] = implode( '<br/>', array_unique( $sources['plugin'] ) );
+						$output[] = '</div>';
+						$output[] = '</div>';
+					}
+					if ( isset( $sources['core'] ) ) {
+						$output[]                  = '<div class="source">';
+						$count                     = count( array_unique( $sources['core'] ) );
+						$sources_container_classes = 'sources-container sources-core';
+						if ( 1 === $count ) {
+							$output[] = sprintf( '<span class="dashicons dashicons-wordpress-alt"></span><strong>%s</strong><br/>', esc_html__( 'Other', 'amp' ) );
+						} else {
+							$output[]                   = sprintf( '<span class="dashicons dashicons-wordpress-alt"></span><strong>%s (%d)</strong><span class="dashicons toggle-sources dashicons-arrow-down"></span>', esc_html__( 'Other', 'amp' ), $count );
+							$sources_container_classes .= ' collapsed';
+						}
+						$output[] = '<div class="' . $sources_container_classes . '">';
+						$output[] = implode( '<br/>', array_unique( $sources['core'] ) );
+						$output[] = '</div>';
+						$output[] = '</div>';
+					}
+					if ( isset( $sources['theme'] ) ) {
+						$output[] = '<div class="source">';
+						$output[] = sprintf( '<span class="dashicons dashicons-admin-appearance"></span><strong>%s</strong>', esc_html( $sources['theme']['name'] ) );
```

Comment:
I'm getting a PHP notice here:

![image](https://user-images.githubusercontent.com/134745/45513216-f200a880-b756-11e8-859a-435fcebbba13.png)

The name isn't stored in the validation error (no should it be). The `$sources['theme']` variable contains an array like `[ 'twentyseventeen' ]`. So this should be looping over each `$theme` and get `wp_get_theme( $theme )->get('Name')`. If the theme does not exist, then the slug should be shown instead (and obviously the result is stale in this case).